[#3983] Add weapon masteries to weapons & characters

arbron opened the pull request:
Adds configuration for weapon masteries and adds them to weapon sheets. Also adds weapon mastery tracking to character data and displays them in the weapons list using a symbol similar to resistance bypasses.

<img width="285" alt="Weapon Mastery - List" src="https://github.com/user-attachments/assets/fbf7a6e9-4363-48e6-b546-a84b2e3e5ae6">

To configure these masteries for a character, this implements a new version of the `TraitSelector` application built in AppV2. This application will hopefully be the basis for new versions of the config applications for actors.

<img width="619" alt="Weapon Mastery - Config" src="https://github.com/user-attachments/assets/b3a4a5cc-4c80-4c23-a8d1-b3f4a49dfb16">

Also adds a `splitSemicolons` utility method to handle splitting semi-colon-separated lists and removing empty entries.

Fyorl commented on module/applications/actor/api/base-config-sheet.mjs:
```diff
@@ -0,0 +1,68 @@
+  async _onRender(context, options) {
+    await super._onRender(context, options);
+    if ( !this.isEditable ) return;
+    for ( const override of this._getActorOverrides() ) {
```
I don't know if we really want to do this in the config sheets. If you have an AE granting you a proficiency, it should appear on your sheet in play mode. If you switch to edit mode to edit the traits, I think it should just show you the source values, never the derived ones.

Fyorl commented on templates/actors/config/weapons-config.hbs:
```diff
@@ -0,0 +1,53 @@
+    <li>
+        <label class="name">{{ localize "DND5E.TraitAll" category=label }}</label>
+        <div class="proficiency">
+            {{ formInput @root.checkbox name=(concat "system.traits.weaponProf.value." @key) value=chosen }}
```
We should use `dnd5e-checkbox`es for these, and below.

```suggestion
            {{ formInput @root.checkbox name=(concat "system.traits.weaponProf.value." @key) value=chosen input=inputs.createCheckboxInput }}
```

```js
// BaseConfigSheet#_prepareContext
context.inputs = { ...foundry.applications.fields, ...dnd5e.applications.fields };
```

Fyorl reviewed (changes requested):
Looks great, thank you Jeff. In addition to my PR comments, I had some small styling comments too:

![image](https://github.com/user-attachments/assets/f78724c4-5329-4457-982d-edabbd21d229)

We can go with the square buttons for our AppV2 sheets (you really hate the circle-clipped document link, huh?), but in that case we should change the Compendium Browser header buttons to match, since that's the only other V2-styled AppV2 app that we have I think. (not required in this PR)

![image](https://github.com/user-attachments/assets/38e261ec-ea45-4f8a-8894-a83eca62ac67)

We should make sure we are using consistent V2 form styles, so this text input, for example, should match the text inputs in the V2 item sheet. I am not actually sure why it's not already. Maybe because it's not in a `.form-group`? We maybe don't need to put the V2 form styles behind the `.form-group` selector.

arbron commented:
> We should make sure we are using consistent V2 form styles, so this text input, for example, should match the text inputs in the V2 item sheet. I am not actually sure why it's not already. Maybe because it's not in a `.form-group`? We maybe don't need to put the V2 form styles behind the `.form-group` selector.

I moved the form styles out of `.form-group`. Had to fix a specificity issue with `item-list-controls`, but that is the only issue I spotted.

Changed over the checkboxes but am having an issue with the checkbox heights when all of the entries are not in the unchecked state:

https://github.com/user-attachments/assets/0e54f79c-ca04-4d02-ba16-7360c91e2f89

Haven't been able to track down exactly what is causing that so if you had any insight it would be welcome.
